# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientInfo.h
# Commit: fb720435215f
# Full Hash: fb720435215fa5be3b2bd2101f3a54382954b70c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1544232
# File Overlap Count: 4
# Description:
#   Bug 1544232 - Limit lifetime extension of SWs by SWs to the sender's lifetime. r=edenchuang
#   
#   This patch introduces an explicit concept of lifetimes with mechanisms
#   in place so that actions taken by Clients (windows or non-ServiceWorker
#   orkers) will extend the lifetime of a ServiceWorker, but a ServiceWorker
# ==============================================================================

diff -r 25452f6623a0 -r fb720435215f dom/clients/manager/ClientInfo.h
--- a/dom/clients/manager/ClientInfo.h	Thu Oct 24 03:02:41 2024 +0000
+++ b/dom/clients/manager/ClientInfo.h	Thu Oct 24 03:02:42 2024 +0000
@@ -37,9 +37,11 @@
   UniquePtr<IPCClientInfo> mData;
 
  public:
-  ClientInfo(const nsID& aId, ClientType aType,
+  ClientInfo(const nsID& aId, const Maybe<nsID>& aAgentClusterId,
+             ClientType aType,
              const mozilla::ipc::PrincipalInfo& aPrincipalInfo,
-             const TimeStamp& aCreationTime);
+             const TimeStamp& aCreationTime, const nsCString& aURL,
+             mozilla::dom::FrameType aFrameType);
 
   ClientInfo(const ClientInfo& aRight);
 